### PR TITLE
Ansible independent upgrade role for all supported component

### PR DESCRIPTION
**Pull request by rajan-mis**
Ansible independent upgrade role for all supported component

Note : Upgrade role is not yet supported for external user and we have not yet added this in the work flow. 

This code is similar to  node role of each supported component. 

Signed-off-by: Rajan Mishra <rajanmis@in.ibm.com>

### CODE REVIEW

File: roles/nfs/upgrade/tasks/install_repository.yml [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  set_fact:
+   scale_nfs_url: 'ganesha_rpms/rhel7/'
+  when: ansible_distribution in scale_rhel_distribution and ansible_distribution_major_version == '7'
+
+- name: upgrade | nfs path
+  set_fact:
+   scale_nfs_url: 'ganesha_rpms/rhel8/'
+  when: ansible_distribution in scale_rhel_distribution and ansible_distribution_major_version == '8'
+
+- name: upgrade | nfs path
```
Comment (Christop1964) [inline roles/nfs/upgrade/tasks/install_repository.yml:21]: The assignment of the path could be wrong when we allow the upgrade on lower releases as well. 
We set the path in other playbooks as follow. 

- name: install | nfs path
  set_fact:
   scale_nfs_url: 'ganesha_debs/ubuntu16/'
  when: ansible_distribution in scale_ubuntu_distribution and ansible_distribution_major_version != '20'

- name: install | nfs path
  set_fact:
   scale_nfs_url: 'ganesha_debs/ubuntu/'
  when: ansible_distribution in scale_ubuntu_distribution and ansible_distribution_major_version == '20'

**Review by Christop1964 — COMMENTED** [review]
So far it look good for me. 
Just added one comment.

File: roles/nfs/upgrade/tasks/install_repository.yml (hunk 1 above)
Comment (rajan-mis) [reply roles/nfs/upgrade/tasks/install_repository.yml:21]: @Christop1964 Thank you. I am planning to add new files that you have changed. Thanks

Comment (rajan-mis) [reply roles/nfs/upgrade/tasks/install_repository.yml:21]: @Christop1964 Done

**Review by Christop1964 — APPROVED** [review]

**Review by mamuthiah — APPROVED** [review]

**Pull request merged**